src/bin/cargo/commands/generate_lockfile.rs

```diff
@@ -9,13 +12,46 @@ pub fn cli() -> Command {
         .arg_manifest_path()
         .arg_lockfile_path()
         .arg_ignore_rust_version_with_help("Ignore `rust-version` specification in packages")
+        .arg(
+            clap::Arg::new("publish-time")
+                .long("publish-time")
+                .value_name("yyyy-mm-ddThh:mm:ssZ")
+                .add(ArgValueCompleter::new(datetime_completer))
+                .help("Latest publish time allowed for registry packages (unstable)")
+                .help_heading(heading::MANIFEST_OPTIONS)
+        )
         .after_help(color_print::cstr!(
             "Run `<bright-cyan,bold>cargo help generate-lockfile</>` for more detailed information.\n"
         ))
 }
 
+fn datetime_completer(current: &std::ffi::OsStr) -> Vec<CompletionCandidate> {
+    let mut completions = vec![];
+    let Some(current) = current.to_str() else {
+        return completions;
+    };
+
+    if current.is_empty() {
+        // While not likely what people want, it can at least give them a starting point to edit
+        let timestamp = jiff::Timestamp::now();
+        completions.push(CompletionCandidate::new(timestamp.to_string()));
+    } else if let Ok(date) = current.parse::<jiff::civil::Date>() {
+        if let Ok(zoned) = jiff::Zoned::default().with().date(date).build() {
+            let timestamp = zoned.timestamp();
+            completions.push(CompletionCandidate::new(timestamp.to_string()));
+        }
+    }
+    completions
+}
```

Comment:
Didn't feel like writing my own parser for the sake of handling incomplete date, so I threw this together to at least make it easier to get the format correct so people can modify them from here.